### PR TITLE
Blog post on the first Duke University meeting

### PR DESCRIPTION
**Pull request by gaurav**
WIP blog post report on the first Duke University meeting in Durham NC from Dec 5-10, 2017. 

PIs: I'd like to go through the text one more time before I ask for reviews from the two of you. I tried to write it more in line with the "report" format we use for meeting reports, but I've still got some "X suggested Y but we decided to do Z" type syntax that I need to fix and some paragraphs that could use better topic sentences. I also haven't been able to get this working in Docker yet (I tested it in plain Jekyll), and would like to give that another shot before I ask for reviews. I should be able to get that done in a day or two!

@GZhang2 Could you please have a look at the post (and especially the pieces about the use cases) and see if I missed anything out? I left out the clade essentialism discussion we had with Hilmar that first morning because I'm hoping you'll expand it into a separate blog post by yourself, but if you think some of that should go in here, please let me know or add it to the pull request!

### CODE REVIEW

**Comment by gaurav** [comment]
Okay, this blog post is now ready for review! You can see [the preview on Github](https://github.com/phyloref/phyloref.github.io/blob/0dc4cc069652176df7b9e3037d52993001f2f7b2/_posts/blog/2018-01-14-first-duke-university-meeting.md). Let me know what you think!

@hlapp I finally got the Docker/Jekyll combination working on my Windows laptop, and have tested this pull request on it! The fix was [surprisingly simple](https://github.com/jekyll/jekyll/issues/5743#issuecomment-271389773).

**Comment by gaurav** [comment]
Okay, this blog post is now ready for review! You can see [the preview on Github](https://github.com/phyloref/phyloref.github.io/blob/0dc4cc069652176df7b9e3037d52993001f2f7b2/_posts/blog/2018-01-14-first-duke-university-meeting.md). Let me know what you think!

@hlapp I finally got the Docker/Jekyll combination working on my Windows laptop, and have tested this pull request on it! The fix was [surprisingly simple](https://github.com/jekyll/jekyll/issues/5743#issuecomment-271389773), and I've set up a separate pull request for it (#32).

**Comment by hlapp** [comment]
> @hlapp I finally got the Docker/Jekyll combination working on my Windows laptop, and have tested this pull request on it! The fix was surprisingly simple, and I've set up a separate pull request for it (#32).
So using the second config file didn't accomplish it already? (A later comment under the link you provided does suggest that it should.)

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  teaser: first-duke-university-meeting-december-2017/duke_chapel_in_snow.jpeg
+  thumb: first-duke-university-meeting-december-2017/duke_chapel_in_snow.jpeg
+---
+
+Since both phyloreferencing postdocs work at the Florida Museum of Natural History with project PI Nico Cellinese, our project proposal calls for us to spend a few weeks every year working closely with our other PI, Hilmar Lapp, at Duke University. We drove up to Durham the week before the [Phenoscape Knowledgebase Datafest and Interoperability Hackathon](http://www.phyloref.org/blog/2017/12/Phenoscape-hackathon/) last month, and spent five days working together to improve our developmental practices, review our software development plan, discuss how to make the value of phyloreferencing clear to potential users, and decide how we would match Phyloreferencing specifiers with nodes on a phylogeny. These discussions followed upon the goals we laid out in our [second face-to-face meeting](http://www.phyloref.org/blog/2017/12/Second-face-to-face-project-meeting/), and helped scope out what we will be working on through the first half of 2018.
```
Comment (hlapp) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: Please put both blog post URLs into the canonical form in which Jekyll renders the base URL.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
...tion](https://github.com/phyloref) on Github and set up a single [organizational repository](https://github.com/phyloref/organization) to store project-wide issues and documents. Our website is itself stored in a [Github repository](https://github.com/phyloref/phyloref.github.io/) as a [Jekyll website](https://jekyllrb.com/). All our software development progresses through pull requests, allowing other project members to review both newly written source code as well as any update to our website, such as new blog posts — for example, [here is the pull request](https://github.com/phyloref/phyloref.github.io/pull/31) for the blog post you are reading right now! Despite taking these steps, we determined that we were failing in two aspects: (1) we were missing a high-level view of our software development efforts, and (2) too many of our documents were created and discussed on websites like Google Docs or Dropbox Paper without being permanently archived in Github or shared publicly through our blog. This is a problem for both project members trying to keep track of everything we are doing as well as for readers of this blog who would like to understand where we are and what we're doing.
```
Comment (hlapp) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: _as well as for readers of this blog who would like to understand_: it's really more about members of the scientific community who may be interested now or some time in the future, than about _current readers_ of the blog. So, for example _as well as people from relevant scientific communities who are, or become,  interested in understanding_.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+To improve these aspects of project management, we decided to increase our use of Github’s [built-in project and document management tools](https://github.com/features/project-management), including:
+
+- Setting up [Kanban boards](https://en.wikipedia.org/wiki/Kanban_(development)) for [the entire project](https://github.com/phyloref/organization/projects/1) (with goals due every two months) and for individual software projects, such as the [Curation Workflow](https://github.com/phyloref/curation-workflow/projects).
+- Move more of our project documents to Github’s Wikis, from [a list of our overall project goals](https://github.com/phyloref/organization/wiki/Project-aims-and-goals) to [weekly progress updates](https://github.com/phyloref/organization/wiki/Goals-and-progress) to records of [casual discussions between team members](https://github.com/phyloref/organization/wiki/Casual-discussions).
```
Comment (hlapp) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: BTW I _think_ you can define custom attributes in the YAML front matter and then use it in the page as `{{ page.my_property }}`. So, you could define https://github.com/phyloref/organization upfront, and then reuse it. Though you've already written it out, so it doesn't matter here.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## A review of our software development plan
+
+One of the most valuable parts of our Durham meeting was the opportunity to discuss our software development plan with [Dan Leehr](https://genome.duke.edu/cores-and-services/computational-solutions/who-we-are), scientific applications architect and senior developer at the [Duke Center for Genomic and Computational Biology](https://genome.duke.edu/), and [Jim Balhoff](https://www.linkedin.com/in/jim-balhoff-30ba415/), senior research scientist at the [Renaissance Computing Institute](http://renci.org/). Dan and Jim were extremely patient while I explained what we were building and how long I thought it would take to build, based on the [software development plan](https://github.com/phyloref/organization/wiki/Software-Development-Plan,-December-2017) I had just written. They thought the timeline I proposed was generally sensible, but that I should schedule more time for some of the more complex software components, which is now reflected in [our six-month plan](https://github.com/phyloref/organization/projects/1).
```
Comment (hlapp) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: _They thought the timeline I proposed was generally sensible, but_ -> I'd focus on what the primary feedback was, which is in the second part of the sentence. (I'm not sure there was enough detail for anyone to pinpoint where and why the proposal timeline wasn't sensible. So focus on what _was_ pointed out.)

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
...s://en.wikipedia.org/wiki/Semantic_Web_Rule_Language) (Semantic Web Rule Language). These could be combined with a fast graph database such as [Blazegraph](https://www.blazegraph.com/), which could store large phylogenies such as the  [Open Tree of Life](https://tree.opentreeoflife.org/about/open-tree-of-life)'s synthetic tree. We might also be able to model phyloreferences in [OWL 2 RL](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/#OWL_2_RL), which would allow rule-based OWL reasoners, such as Jim’s [Arachne](https://github.com/balhoff/arachne), to process them more efficiently than full OWL 2 reasoners. [OWL 2 EL](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/#OWL_2_EL) reasoners, like [ELK](https://github.com/liveontologies/elk-reasoner), cannot fully represent the phyloreferencing model we currently use. Restricting our model to the OWL 2 RL or OWL 2 EL profiles might require us to make some counter-intuitive modeling decisions, such as modeling phylogenetic nodes as classes rather than individuals, but could provide us with speed improvements and allow us to use faster reasoners. We will reconsider these options as we curate more phyloreferences in the future.
```
Comment (hlapp) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: _full OWL 2 reasoners_ -> You probably mean full OWL2-DL reasoners here? That's quite an important difference ...

**Comment by gaurav** [comment]
@hlapp Nope, using the second config file didn't work! The specific problem I was having was that Jekyll was writing out URLs in the HTML output as "http://0.0.0.0:4000/...". I could access "http://127.0.0.1:4000/" in my browser without any problem, but all of the links, images or CSS files on that page failed since my browser was trying and failing to connect to 0.0.0.0 (I wonder if that's a Windows specific thing, and if "http://0.0.0.0/" just works in Linux?). 

If I changed the Jekyll "host" setting to "127.0.0.1" or "localhost", I could no longer connect to "http://127.0.0.1:4000/" in my browser at all. Setting the environmental variable as suggested in the linked thread removes the site name from URLs in the the page entirely, so all the links become relative to the domain root (e.g. "/css/main.css"), and everything works fine.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 1 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Do you mean `{{site.baseurl}}{% post_url /blog/2017-12-30-Phenoscape-hackathon %}`? Done!

It also worked without the `/blog/`, but produced the following warning:

> Deprecation: A call to '{% post_url 2017-12-30-Phenoscape-hackathon %}' did not match a post using the new matching method of checking name (path-date-slug) equality. Please make sure that you change this tag to match the post's name exactly.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 2 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Fixed to: "This is a problem for project members trying to keep track of our current progress as well as for people from relevant scientific communities who are, or become, interested in understanding what we have built and what we’re working on right now."

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 3 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: I think I prefer making the URLs clear and explicit in each case -- there's a greater possibility of messing up a URL (which is why I test them all!), but I think it'll be harder to tell what's going on just by reading the source code. You can convert from one form to the other pretty easily with regular expressions anyway :).

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 4 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: 👍  Now changed to "They recommended that I schedule more time for some of the more complex software components, which is now reflected in our six-month plan."

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 5 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Oh huh, I thought they were the same thing! I've changed this to "OWL 2 DL" and added a link to https://en.wikipedia.org/wiki/Web_Ontology_Language#OWL_DL.

**Comment by gaurav** [comment]
Thanks, Hilmar! I'll check with Guanyang and Nico at our phone meeting tomorrow before squash-and-merging this pull request.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+While we work to identify these use cases by reaching out to potential collaborators, we need a tool that allows the basic use of phyloreferences to be demonstrated. The first step towards this goal will be to build the [Curation Tool](https://github.com/phyloref/curation-tool), which will our first user-accessible tool for writing and executing phyloreferences. This will allow project members to play with the current phyloreferencing model in practice and identify gaps that might not be obvious just from [reading the specification](https://github.com/phyloref/specification/blob/master/specification.md). It will be the first tool we can show to potential collaborators, and any custom tools necessary for their use can be based on it. The creation of the Curation Tool is therefore our immediate software development goal.
```
Comment (GZhang2) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: missing a word here: which will [be] our first user-accessible tool

File: _posts/blog/2018-01-14-first-duke-university-meeting.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+## How to match Phyloreferencing specifiers with nodes on a phylogeny
+
+My most important goal for this meeting was to finalize a model we could use to match Phyloreferencing specifiers. This is the first step in resolving phyloreferences, in which “specifiers” — the parts of a phyloreference that refer to biological entities, such as “*Homo sapiens*” or “Gaurav Vaidya” — are matched with nodes on a phylogeny that refer to those entities. This matching process can be exceptionally complex, and so we quickly decided not to carry it out purely in OWL. If we carried out the matching in a programming language or SPARQL, we could more easily write, maintain and extend the logic necessary to carry out matching. We are currently updating the Phyloreferencing specification to reflect this decision, and will shortly publish a blog post detailing the specifier model we have settled on.
```
Comment (GZhang2) [inline _posts/blog/2018-01-14-first-duke-university-meeting.md]: I find it odd to use a human name as a specifier in the context of biological research.

**Comment by GZhang2** [comment]
Hi Gaurav, I've read the draft. I am largely in agreement with you and made just two minor comments regarding the use case part. Thanks for writing this.  - Guanyang

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 7 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Would "Specimen FMNH PR 2081" work instead? I wanted to find a specimen specifier that I could casually throw in without confusing people.

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 6 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Good catch - fixed!

File: _posts/blog/2018-01-14-first-duke-university-meeting.md (hunk 7 above)
Comment (gaurav) [reply _posts/blog/2018-01-14-first-duke-university-meeting.md]: Going with [CM 9380](https://en.wikipedia.org/wiki/Specimens_of_Tyrannosaurus#Holotype:_CM_9380) as that's at least a holotype.

**Comment by gaurav** [comment]
Thanks, @GZhang2! I'll reread it one last time and publish it later today.

**Comment by hlapp** [comment]
@gaurav I needed this up for the email to the AB. You're welcome to still make small changes (and I don't think you were considering major ones?), just create another PR (so we get notified).

**Pull request merged**